source/language/types.rst

```diff
@@ -279,25 +279,53 @@ their argument expression in parentheses:
 
    .. table:: Built-in mathematical functions in OpenQASM3.
 
-      +--------+
-      | arccos |
-      +--------+
-      | arcsin |
-      +--------+
-      | arctan |
-      +--------+
-      | cos    |
-      +--------+
-      | exp    |
-      +--------+
-      | ln     |
-      +--------+
-      | sin    |
-      +--------+
-      | sqrt   |
-      +--------+
-      | tan    |
-      +--------+
+      +----------+--------------------------------+---------------------+
+      | Function | [Input Range/Type] , [...]     | [Output Range/Type] |
+      +==========+================================+=====================+
+      | arccos   | [-1, 1]                        | [0, pi rad]         |
+      +----------+--------------------------------+---------------------+
+      | arcsin   | [-1, 1]                        | [-pi/2, pi/2 rad]   |
+      +----------+--------------------------------+---------------------+
+      | arctan   | [-∞, Real, ∞]                  | [-pi/2, pi/2 rad]   |
+      +----------+--------------------------------+---------------------+
+      | ceiling  | [float], [float]               | [float]             |
+      +----------+--------------------------------+---------------------+
+      | cos      | [float]                        | [float]             |
+      +----------+--------------------------------+---------------------+
+      | div      | ??                             | ??                  |
+      +----------+--------------------------------+---------------------+
+      | exp      | [float]                        | [float]             |
+      |          |                                |                     |
+      |          | [complex]                      | [complex]           |
+      +----------+--------------------------------+---------------------+
+      | floor    | [float], [float]               | [float]             |
+      +----------+--------------------------------+---------------------+
+      | log *(e)*| [float]                        | [float]             |
+      +----------+--------------------------------+---------------------+
+      |  mod     | [int], [int]                   | [int]               |
+      |          |                                |                     |
+      |          | [float], [int,float]           | [float]             |
+      +----------+--------------------------------+---------------------+
+      | popcount | [bits, unsigned]               | [positive int]      |
+      +----------+--------------------------------+---------------------+
+      | pow      | [int], [unsigned int]          | [int]               |
+      |          |                                |                     |
+      |          | [float], [int,float]           | [float]             |
+      |          |                                |                     |
+      |          | [complex], [int,float,complex] | [complex]           |
```

Comment:
This is another one we'll need to specify the principal root for, but this table probably isn't the right place to do it.

Reply:
Not sure I understand your comment.

Reply:
`pow` can take a fractional power when the right-hand value is a float, so it's subject to the same concerns as `sqrt` about defining which value is returned.  For example, `pow(x, 0.5)` is (mathematically) equivalent to `sqrt(x)`, so we should specify the value returned.  Roughly, in the situation that there's several possible solutions, we return the one which has the largest positive real part, and break ties by picking the one with the positive imaginary part.  I need to think a little more about how that works with regards to complex powers, but I think the definition extends fairly easily - it's the same that I imagine all programming languages use.

Reply:
We also need to spec `div`. I left that an open question.